folding-schemes/src/folding/protogalaxy/folding.rs

```diff
@@ -473,14 +503,17 @@ mod tests {
             phi,
             betas: betas.clone(),
             e: Fr::zero(),
+            u,
+            x,
         };
         // same for the other instances
         let mut witnesses: Vec<Witness<Fr>> = Vec::new();
         let mut instances: Vec<CommittedInstance<Projective>> = Vec::new();
         #[allow(clippy::needless_range_loop)]
         for i in 0..k {
+            let (u_i, x_i, w_i) = get_test_z_split::<Fr>(i + 4);
```

Comment:
Is that 4 comming from ` 1 + x.len() + w.len();`?? Why not using `n` then?

Reply:
That's the `input` param used to generate the witness (ie. the `x` variable used in the equation `y=x^3+x+5` constrainted in the circuit, so that each iteration generates different witnesses (for different given `x`s). The `+4` is just to not start at `0` or at `1`, could be any arbitrary number.

Reply:
Sorry for the late response!

As @arnaucube mentioned, `4` here can be arbitrary value. IIRC I just copied this line from here, https://github.com/privacy-scaling-explorations/sonobe/blob/edadcdd520b8cf657cfa5c679dd09c6579759b0c/folding-schemes/src/folding/protogalaxy/folding.rs#L453-L458 without further checking the details. Let me refactor this and generate the inputs from `rng` instead of using magic numbers!

Reply:
Done 😄